jira_release/jira_release.py

```diff
@@ -154,6 +162,82 @@ def comment_and_close_issues_to_deploy(
             self.__comment_confirm_deploy(environment, issue)
             self.__close_and_resolve(issue)
 
+    def generate_changelog(
+        self,
+        jira_prefix,
+        remote_version,
+        to_deploy_version="HEAD",
+        git_path=".",
+        changelog_path="CHANGELOG.md",
+        initialize=False,
+    ):
+        """
+        Generates/updates a changelog data file found in git_path, based on JIRA issues to be deployed
+        @param jira_prefix: The prefix of the JIRA issue
+        @param remote_version: The version currently live on the remote server
+        @param to_deploy_version: The version to be deployed
+        @param git_path: The path of the local git repository
+        @param changelog_path: The relative path of the changelog file within git_path
+        @param initialize: Creates or overwrite the changelog file
+        @return: None
+        """
+
+        full_changelog_path = os.path.join(git_path, changelog_path)
+        changelog_gen = Changelog(self.jira, full_changelog_path)
+
+        try:
+            file_exists = changelog_gen.check_if_changelog_exists(raise_exception=False)
+            if initialize:
+                if file_exists and (
+                    input(
+                        f"A changelog file already exists, do you want to overwrite it? [y/N]: "
+                    ).lower()
+                    != "y"
+                ):
+                    return
+                has_to_initialize = True
+            elif not file_exists:
+                if (
+                    input(
+                        f"A changelog file was not found. Would you like to initialize it ? [y/N]: "
+                    ).lower()
+                    != "y"
+                ):
+                    return
+                else:
+                    has_to_initialize = True
+            else:
+                has_to_initialize = False
+
+            if file_exists and not has_to_initialize:
+                first_git_marker = changelog_gen.find_first_git_marker()
+                if remote_version != first_git_marker and (
+                    input(
+                        f"WARNING: The remote version is different from the last deployment found in {changelog_path} !\n"
```

Comment:
As far as I understand, this warning (in its current form) will be a false positive most of the time because a changelog generation will be followed by a commit for the changelog itself (and this commit cannot update the git marker with itself).

Reply:
Any ideas to solve this while keeping the warning? Maybe check the diffs between the `remote_version`  and the git marker?